packages/docs/composition/getters.md

```diff
@@ -1,69 +1,88 @@
-Each composable has its own dedicated getter, e.g., `useCart` and `cartGetters`. **You should only use getters dedicated to specific composable**.
+Reading the same composable's data in a few places in your application can be cumbersome, especially if that data is deeply nested.
 
-## When should I use them?
+That's why every composable has a dedicated getter, which is a helper for extracting commonly used data. For example, the `useUser` composable has the `userGetters`.
 
-**Getters should be used whenever possible to extract the data from composables**, e.g., when you want to get the quantity of the products in the cart in UI components.
+You should use getters whenever possible unless the given getter doesn't have a method that returns the data you need.
 
-## How can I use getters?
+## Anatomy of a getter
 
-The getters accept arguments to get the data. As an example, we will use cart functionalities.
-Most functions in `cartGetters` accept whole `cart` objects or individual cart items. This data needs to be extracted from `useCart` composable first.
-As the first example, we will use `getTotalItems` from `cartGetters` to get the total number of items in the cart.
+Getters are objects with a set of methods that accept either:
 
-```vue
-<template>
-  <div>
-    <!-- ... -->
-    <span>
-      {{ cartTotalItems }}
-    </span>
-  </div>
-</template>
+- the [primary state](./composables.html#anatomy-of-a-composable) from their composable,
+- or result from other getter methods.
+
+Let's take a closer look at how it might look like using the [userGetters](/reference/api/core.usergetters.html) as an example:
+
+<img
+  src="../images/userGetters-getter-anatomy.webp"
+  alt="Anatomy of the userGetters getter"
+  style="display: block; margin: 0 auto;">
+
+In this example, the getter has four methods:
 
+- `getEmailAddress()`,
+- `getFirstName()`,
+- `getFullName()`,
+- `getLastName()`.
+
+Each accepts the `user` object from the `useUser` composable as a parameter.
+
+## Usage
+
+Let's see how you can use the [userGetters](/reference/api/core.usergetters.html) to get the user's full name:
+
+```vue
 <script>
-import { computed } from '@nuxtjs/composition-api';
-import { useCart, cartGetters } from '{INTEGRATION}';
-import { onSSR } from '@vue-storefront/core';
+import { useUser, userGetters } from '{INTEGRATION}';
+import { useFetch, computed } from '@nuxtjs/composition-api';
 
 export default {
   setup() {
-    const { cart, load } = useCart();
-    const cartTotalItems = computed(() => cartGetters.getTotalItems(cart.value);
+    const { load, user } = useUser();
+    const userFullName = computed(() => userGetters.getFullName(user.value));
 
-    onSSR(async () => {
+    useFetch(async () => {
       await load();
     });
 
     return {
-      cart,
-      cartTotalItems,
+      userFullName
     };
   }
-}
+};
```

Comment:
Just an idea, but it might be helpful to have code comments that describe the steps. It would put the explanation directly with the associated line.

Open to hearing your thoughts.
```
    const { load, user } = useUser();
    
    // gets the full name from `user`
    const userFullName = computed(() => userGetters.getFullName(user.value));

    // will update `user` if successful
    useFetch(async () => {
      await load();
    });

    // make `userFullName` available in our template
    return {
      userFullName
    };
  

```

Reply:
Done in https://github.com/vuestorefront/vue-storefront/pull/6760/commits/5163869dfa088837388d950516ce571c0e18a74f